vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/ValidationController.java

```diff
@@ -125,4 +68,18 @@ public void validate(V value) {
             setValidationResultErrorMessage("");
         }
     }
+
+    private void setInvalid(boolean invalid) {
+        component.setInvalid(invalid);
+    }
+
+    private void setValidationResultErrorMessage(String errorMessage) {
+        if (component.getErrorMessage() == null
+                || component.getErrorMessage().isEmpty()
+                || component.getErrorMessage()
+                        .equals(validationResultErrorMessage)) {
```

Comment:
It's not immediately obvious that this is used to detect whether the current error message is a custom one or from a constraint. Maybe:
- Rename `validationResultErrorMessage` to `lastValidationResultErrorMessage`
- Extract the condition into a variable with a helpful name, or add a comment explaining what this check does

Reply:
Thanks, done.